README.markdown

```diff
@@ -52,72 +76,206 @@ class app_widgetContainer {
+
 ### Enumerations
 
-Use lowerCamelCase for enumeration values:
+Following Apple's API Design Guidelines for Swift 3, use lowerCamelCase for enumeration values.
 
 ```swift
 enum Shape {
     case rectangle
     case square
-    case triangle
-    case circle
+    case rightTriangle
     case equilateralTriangle
 }
 ```
 
 ### Prose
 
-When referring to functions in prose (tutorials, books, comments) include the required parameter names from the caller's perspective or `_` for unnamed parameters.
+When referring to functions in prose (tutorials, books, comments) include the required parameter names from the caller's perspective or `_` for unnamed parameters. Examples:
 
 > Call `convertPointAt(column:row:)` from your own `init` implementation.
 >
 > If you call `dateFromString(_:)` make sure that you provide a string with the format "yyyy-MM-dd".
 >
-> If you call `timedAction(delay:perform:)` from `viewDidLoad()` remember to provide an adjusted delay value and an action to perform.
+> If you call `timedAction(afterDelay:perform:)` from `viewDidLoad()` remember to provide an adjusted delay value and an action to perform.
 >
 > You shouldn't call the data source method `tableView(_:cellForRowAtIndexPath:)` directly.
 
-When in doubt, look at how Xcode lists the method in the jump bar – our style here matches that.
+This is the same as the `#selector` syntax. When in doubt, look at how Xcode lists the method in the jump bar – our style here matches that.
 
 ![Methods in Xcode jump bar](screens/xcode-jump-bar.png)
 
+
 ### Class Prefixes
 
-Swift types are automatically namespaced by the module that contains them and you should not add a class prefix. If two names from different modules collide you can disambiguate by prefixing the type name with the module name.
+Swift types are automatically namespaced by the module that contains them and you should not add a class prefix such as RW. If two names from different modules collide you can disambiguate by prefixing the type name with the module name. However, only specify the module name when there is possibility for confusion which should be rare.
 
 ```swift
 import SomeModule
 
 let myClass = MyModule.UsefulClass()
 ```
 
+### Selectors
+
+Selectors are Obj-C methods that act as handlers for many Cocoa and Cocoa Touch APIs. Prior to Swift 2.2, they were specified using type unsafe strings. This now causes a compiler warning. The "Fix it" button replaces these strings with the **fully qualified** type safe selector. Often, however, you can use context to shorten the expression. This is the preferred style.
+
+**Preferred:**
+```swift
+let sel = #selector(viewDidLoad)
+```
+
+**Not Preferred:**
+```swift
+let sel = #selector(ViewController.viewDidLoad)
+```
+
```

Comment:
What's the rest of the ...? I'm assuming the selector call here is just ```#selector(Selector.viewDidLoad)```? I'm actually fine not defining this section at all if there are various ways of doing this that are "not preferred"